pkg/kubelet/prober/manager.go

```diff
@@ -207,6 +207,23 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *api.PodStatus) {
 		}
 		podStatus.ContainerStatuses[i].Ready = ready
 	}
+	// init containers are ready if they have exited with success or if a readiness probe has
+	// succeeded.
+	for i, c := range podStatus.InitContainerStatuses {
```

Comment:
My concern is more that drift between the two types of containers is really
hard unless everything in the kubelet is identical (shared code is >>> not
shared code).  Since the API enforces those fields to be unset, I think
having two paths in a lot of places in the kubelet is going to be worse.

On Thu, May 5, 2016 at 2:46 PM, Prashanth B <notifications@github.com>
wrote:

> In pkg/kubelet/prober/manager.go
> <https://github.com/kubernetes/kubernetes/pull/23567#discussion_r62234597>
> :
>
> > @@ -207,6 +207,23 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *api.PodStatus) {
> >  		}
> >  		podStatus.ContainerStatuses[i].Ready = ready
> >  	}
> > +	// init containers are ready if they have exited with success or if a readiness probe has
> > +	// succeeded.
> > +	for i, c := range podStatus.InitContainerStatuses {
>
> Remove dead code? Or is it actually required for something higher up the
> stack to not barf, if so suggest a TODO.
>
> It looks like the proposal leaves readiness as an open question, so
> leaving it in is confusing, especially if it needs to be preserved across a
> refactor.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23567/files/9813a4db58c159c5ad4f74de7e9e4fd3c0002957#r62234597>
>